packages/fxa-settings/src/components/PageDeleteAccount/index.tsx

```diff
@@ -0,0 +1,221 @@
+    setConfirmed(true);
+
+    logViewEvent('flow.settings.account-delete', 'terms-checked.success');
+  };
+
+  const deleteAccount = useAccountDestroyer({
+    onSuccess: () => {
+      setAccountDeleted(true);
+
+      logViewEvent('flow.settings.account-delete', 'confirm-password.success');
+    },
+    onError: (error) => {
+      if (error.errno === 103) {
+        setErrorText(error.message);
+        setValue('password', '');
+      } else {
+        alertBar.setType('error');
+        alertBar.setContent(error.message);
+        alertBar.show();
+        logViewEvent('flow.settings.account-delete', 'confirm-password.fail');
+      }
+    },
+  });
+
+  const onFormSubmit = ({ password }: FormData) => {
+    deleteAccount.execute(primaryEmail.email, password, sessionToken()!);
+  };
+
+  return (
+    <>
+      {accountDeleted ? (
+        <FlowContainer title="Delete Account">
+          Your account has been deleted
+        </FlowContainer>
+      ) : (
+        <FlowContainer title="Delete Account" subtitle={subtitleText}>
+          {alertBar.visible && (
+            <AlertBar onDismiss={alertBar.hide} type={alertBar.type}>
+              <p data-testid="delete-account-error">{alertBar.content}</p>
+            </AlertBar>
+          )}
+          <VerifiedSessionGuard onDismiss={goBack} onError={goBack} />
+          {!confirmed && (
+            <div className="my-4 text-sm" data-testid="delete-account-confirm">
+              <p className="mb-4">
+                You've connected your Firefox account to Mozilla products that
+                keep you secure and productive on the web:
+              </p>
+              <div className="p-2">
+                <ul className="list-inside mb-4">
+                  <li className="list-disc">
+                    <a className="link-blue" href="https://vpn.mozilla.org/">
+                      Mozilla VPN
+                    </a>
+                  </li>
+                  <li className="list-disc">
+                    <a
+                      className="link-blue"
+                      href="https://www.mozilla.org/en-US/firefox/lockwise/"
+                    >
+                      Firefox Lockwise
+                    </a>
+                  </li>
+                  <li className="list-disc">
+                    <a
+                      className="link-blue"
+                      href="https://monitor.firefox.com/"
+                    >
+                      Firefox Monitor
+                    </a>
+                  </li>
+                </ul>
+              </div>
+              <p className="mb-4">
+                Please acknowledge that by deleting your account:
+              </p>
+              <form
+                onChange={(evt: FormEvent<HTMLFormElement>) => {
+                  setHideContinueButton(
+                    evt.currentTarget.querySelectorAll('input:checked').length <
+                      4
+                  );
```

Comment:
This is an optional suggestion based on something I did over in Experimenter, but might yield an cleaner setup for these checkboxes:

- Set up [an array](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/FormRequestReview/index.tsx#L14-L17) of label strings.
- [Iteratively](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/FormRequestReview/index.tsx#L51-L64) render them (hooray DRY)
- When one is checked, [add/remove](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/FormRequestReview/index.tsx#L29-L40) it from an internal state
- On render, [compare](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/FormRequestReview/index.tsx#L25-L27) that internal state against the original array of labels to see if they match, and assign the result to `hideContinueButton`

I don't feel strongly about this, just a thought since it helps avoid repetition and needing to inspect the DOM.